pyxem/utils/ri_utils.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2019 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+
+import numpy as np
+from diffsims.utils.atomic_scattering_params import ATOMIC_SCATTERING_PARAMS
+from diffsims.utils.lobato_scattering_params import ATOMIC_SCATTERING_PARAMS_LOBATO
+
+
+def scattering_to_signal(elements, fracs, N, C, s_size, s_scale, type='lobato'):
+    """ A function to override HyperSpy's as_signal method in a fit, as that
```

Comment:
This makes me nervous, are you sure we need to do this?

Reply:
I would rather not, but the multifit method was consistently failing, requiring me to write a workaround. I spent about 3 or 4 days on this in February and found nothing that worked bar a workaround.

Reply:
I think it's worth having another look - hyperspy may have changed since then. If it hasn't then please can you document the errors you get and submit an issue to hyperspy because this isn't great....

Reply:
In fact, this may be resolved by the use of the new hyperspy_extensions features!

You should add your components to the hyperspy_extenstion.yaml (https://github.com/pyxem/pyxem/blob/master/pyxem/hyperspy_extension.yaml)